builder/image_solver.go

```diff
@@ -198,24 +198,36 @@ func (m *multiImageSolver) SolveImages(ctx context.Context, imageDefs []*states.
 		return ret, nil
 	}
 
+	newInterImgFormat := false
+	info, err := m.bkClient.Info(ctx)
+	if err != nil {
+		// Maybe older buildkit.
+	} else {
+		switch info.BuildkitVersion.Version {
+		case "v0.6.19":
+		case "v0.6.20":
+		default:
+			newInterImgFormat = true
+		}
+	}
+
+	onPullMap := make(map[string]string) // intermediate image name -> final image name
```

Comment:
is this duplicating https://github.com/earthly/earthly/blob/main/util/gatewaycrafter/pullpingmap.go ?

we have a similar onPull function under builder, where we pull images out of the pullpingmap: https://github.com/earthly/earthly/blob/main/builder/builder.go#L412-L416

Reply:
It is very similar but different enough. The intention is the same - to keep track of a mapping between intermediate names vs final (re-tagged) names. However, the images are not exported locally - they are exported into `WITH DOCKER` - so the code you're referencing couldn't be reused in this situation.